crates/sequencer/src/runner.rs

```diff
@@ -302,37 +299,69 @@ where
-                                    evm.get_last_pending_transaction(&mut working_set_to_discard);
-
-                                if let Some(last_tx) = last_tx {
-                                    if last_tx.hash() == *evm_tx.hash() {
-                                        all_txs.push(rlp_tx);
-                                    }
+                                    &mut working_set,
+                                ) {
+                                    Ok(result) => result,
+                                    Err(e) => match e {
+                                        // Since this is the sequencer, it should never get a soft confirmation error or a hook error
+                                        sov_rollup_interface::stf::StateTransitionError::SoftConfirmationError(soft_confirmation_error) => panic!("Soft confirmation error: {:?}", soft_confirmation_error),
+                                        sov_rollup_interface::stf::StateTransitionError::HookError(soft_confirmation_hook_error) => panic!("Hook error: {:?}", soft_confirmation_hook_error),
+                                        sov_rollup_interface::stf::StateTransitionError::ModuleCallError(soft_confirmation_module_call_error) => match soft_confirmation_module_call_error {
+                                            // if we are exceeding block gas limit with a transaction 
+                                            // we should inspect the gas usage and act accordingly
+                                            // if there is room for another transaction
+                                            // keep trying txs
+                                            // if not, break
+                                            sov_modules_api::SoftConfirmationModuleCallError::EvmGasUsedExceedsBlockGasLimit {
+                                                cumulative_gas,
+                                                tx_gas_used: _,
+                                                block_gas_limit
+                                            } => {
+                                               if block_gas_limit - cumulative_gas < MIN_TRANSACTION_GAS {
+                                                break;
+                                               } else {
+                                                working_set_to_discard = working_set.revert().to_revertable();
+                                                continue;
+                                               }
+                                            },
+                                            // we configure mempool to never accept blob transactions
+                                            // to mitigate potential bugs in reth-mempool we should look into continue instead of panicking here
+                                            sov_modules_api::SoftConfirmationModuleCallError::EvmBlobGasUsedExceedsBlockGasLimit => panic!("got blob tx from mempool"),
```

Comment:
Why is this a panic?

Reply:
tbh I panicked bc for the sequencer, getting a hook error or soft confirmation error meant there is something fundamentally wrong with itself. if everybody says we shouldn't, then we can propagate error upstream

Reply:
I agree with panic here